ci.jenkins.io.tf

```diff
@@ -0,0 +1,93 @@
+# Service: ci.jenkins.io
+
+##########################################################################################################################################################
+## Section: S3 Bucket used for storing Artifact and stashes
+## This bucket does not need logging, versionning nor encryption as all objects are public
+#tfsec:ignore:aws-s3-enable-versioning tfsec:ignore:aws-s3-enable-bucket-logging tfsec:ignore:aws-s3-encryption-customer-key tfsec:ignore:aws-s3-enable-bucket-encryption
+resource "aws_s3_bucket" "ci_jenkins_io_artifacts" {
+  bucket = "ci-jenkins-io-artifacts"
+
+  force_destroy = true
+
+  tags = {
+    jenkins = "ci.jenkins.io"
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "ci_jenkins_io_artifacts" {
+  bucket                  = aws_s3_bucket.ci_jenkins_io_artifacts.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_iam_user" "ci_jenkins_io_artifacts" {
+  name = "ci-jenkins-io-artifacts"
+
+  tags = {
+    jenkins = "ci.jenkins.io"
+  }
+}
+
+resource "aws_iam_access_key" "ci_jenkins_io_artifacts" {
+  user = aws_iam_user.ci_jenkins_io_artifacts.name
+  # No pgp_key provided: the secret value is unencrypted in the state file (which is fine: we encrypt the state file here with sops)
+}
+
+resource "aws_iam_policy" "ci_jenkins_io_artifacts" {
+  name        = "ci-jenkins-io-artifacts"
+  description = "S3 Artifact Manager for ci.jenkins.io"
+
+  policy = data.aws_iam_policy_document.ci_jenkins_io_artifacts_iam.json
+}
+
+data "aws_iam_policy_document" "ci_jenkins_io_artifacts_iam" {
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = [aws_s3_bucket.ci_jenkins_io_artifacts.arn]
+    effect    = "Allow"
+  }
+  statement {
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:DeleteObject",
```

Comment:
The 2 policies documents are for 2 "layers":

- The first is for the IAM permissions (e.g. AWS API permissions) and needs to be restricted to the S3 bucket's ARN (as we don't want ci.jenkins.io to see other buckets if any)
- The 2nd is a S3 policy level (e.g. inside the 2S api itself)


Both were required to not have an HTTP/403 error (first one was catcheable in the general configuration, the 2nd one when the real stash/unstash operation happenned). I guess it's because of https://aws.amazon.com/blogs/aws/heads-up-amazon-s3-security-changes-are-coming-in-april-of-2023/